client/packages/theia-integration/src/browser/language/glsp-client-contribution.ts

```diff
@@ -105,16 +106,44 @@ export abstract class BaseGLSPClientContribution implements GLSPClientContributi
         }));
     }
     protected readonly toDeactivate = new DisposableCollection();
+
     activate(): Disposable {
         if (this.toDeactivate.disposed) {
-            this.doActivate(this.toDeactivate);
+            this.doActivate(this.toDeactivate)
+                .then(() => this.initialize());
         }
         return this.toDeactivate;
     }
+
     deactivate(): void {
         this.toDeactivate.dispose();
     }
 
+    protected createInitializeParameters(): InitializeParameters {
+        const initOptions = this.createInitializeOptions();
+        return initOptions ? { options: initOptions } : {};
```

Comment:
FYI: This could be simplified by reusing the interface key name for the const:
<code>
         const options = this.createInitializeOptions();
        return { options };
</code>